dtbase/tests/test_api_sensors.py

```diff
@@ -245,11 +248,33 @@ def test_delete_sensor(client: FlaskClient) -> None:
 
 
 @pytest.mark.skipif(not DOCKER_RUNNING, reason="requires docker")
-def test_delete_sensor_type(client: FlaskClient) -> None:
-    with client:
+def test_delete_sensor_type(auth_client: AuthenticatedClient) -> None:
+    with auth_client as client:
         response = insert_weather_type(client)
         assert response.status_code == 201
         response = client.delete(
             "/sensor/delete-sensor-type", json={"type_name": "weather"}
         )
         assert response.status_code == 200
+
+
+@pytest.mark.skipif(not DOCKER_RUNNING, reason="requires docker")
+def test_unauthorized(client: FlaskClient) -> None:
+    """Check that we aren't able to access any of the end points if we don't have an
+    authorization token.
+
+    Note that this one, unlike all the others, uses the `client` rather than the
+    `auth_client` fixture.
+    """
+    with client:
```

Comment:
This is essentially a list of all our pages right? Because none should be accessible without the password? 

Do we have a way of collecting all our pages automatically? I can see us forgetting to add pages to this test as we create more. .

Reply:
Oh sorry, not the pages, all the sensor endpoints.

Reply:
Yeah, that's all the endpoints except for `/auth/login`. Automated discovery would indeed be better. Not sure how to do it, but I'm sure something could be worked out.

Reply:
Nice, thanks! I added the same automation for `users`, `models`, and `locations` too.